src/Data/Auth/AuthProvider.tsx

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import { AuthConfig, KeycloakProvider, LocalConfig } from "./KeycloakProvider";
+import { DatabaseAuthProvider } from "./PartialProviders/DatabaseAuthProvider";
+import { KeycloakAuthProvider } from "./PartialProviders/KeycloakAuthProvider";
+import { NoAuthProvider } from "./PartialProviders/NoAuthProvider";
+
+interface Props {
+  config: undefined | AuthConfig | LocalConfig;
+}
+
+/**
+ * AuthProvider component renders the appropriate authentication provider based on the provided config
+ *
+ * @param config - The authentication configuration object sent from backend it includes type of authorization that should be initialized, and in case of keycloak authorization its required config {KeycloakConfig}.
+ * @returns The rendered authentication provider component.
+ */
+export const AuthProvider: React.FC<React.PropsWithChildren<Props>> = ({
+  children,
+  config,
+}) => {
+  switch (config?.method) {
+    case "database":
+      return <DatabaseAuthProvider>{children}</DatabaseAuthProvider>;
+    case "oidc":
+      return (
+        <KeycloakProvider config={config}>
```

Comment:
This doesn't make sense to me. Any reason why you wrapped it like this instead of having the entire implementation in one file?

Reply:
I tried it, but when it was in one file I couldn't reach the keycloak instance from the keycloakprovider. The logic has to be one level below initialisation of the context I guess

Reply:
A full application in React could hypothetically be written in one single component, we are splitting it up for readability and organization, but technically speaking, everything that can be achieved with multiple components can be done in one. 

If you can't figure it out, which is fine, make a ticket for it, and at least, place all the elements/wrappers/parts concerning Keycloak in on single file. You should only call one Provider for Keycloak here.

Reply:
Of course it could be done, but then there would be no need for useContext. And as far as I know and searched context provider has to be one level higher that the component where we want to use given context.

I'll merge the files.